src/test/java/core/basesyntax/strategy/impl/BalanceOperationTest.java

```diff
@@ -0,0 +1,47 @@
+package core.basesyntax.strategy.impl;
+
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.db.StorageImpl;
+import core.basesyntax.exception.InvalidValueExeption;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationProcessor;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class BalanceOperationTest {
+    private static Storage storageImpl;
+    private static OperationProcessor balanceOperation;
+
+    @BeforeAll
+    static void setUp() {
+        storageImpl = new StorageImpl();
+        balanceOperation = new BalanceOperation((StorageImpl) storageImpl);

```

Comment:
The root cause of why you need to cast your storage explicitly is because your Operation classes accept StorageImpl instance and not Storage one which is technically violation of Dependency Invertion Principle, so consider refactoring your code so there is no need for workarounds like this